creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -377,37 +388,73 @@ class SnapbackSM {
-
+    let issueReconfig = false
+    let newPrimary, newSecondary1, newSecondary2
+    while (newReplicaNodesSet.size !== NUMBER_OF_REPLICA_SET_NODES && i++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
       // Select a random node that is not from the current replica set
-      newSecondary = healthyPotentialPeers[Math.floor(Math.random() * healthyPotentialPeers.length)]
+      let randomHealthyNode = healthyNodes[Math.floor(Math.random() * healthyNodes.length)]
+
+      // If node is already present in set, keep finding a unique healthy node
+      if (newReplicaNodesSet.has(randomHealthyNode)) continue
 
       // Check to make sure that the newly selected secondary does not have existing user state
       try {
-        const clockValue = await CreatorNode.getClockValue(newSecondary, wallet)
-        if (clockValue === -1) currentNodeHasExistingUserState = false
+        const clockValue = await CreatorNode.getClockValue(randomHealthyNode, wallet)
+        if (clockValue === -1) {
+          newReplicaNodesSet.add(randomHealthyNode)
+        }
       } catch (e) {
         // Something went wrong in checking clock value. Reselect another secondary.
-        this.logError(e.message)
+        this.logError(`[determineNewReplicaSet] ${e.message}`)
       }
     }
 
-    if (currentNodeHasExistingUserState) throw new Error(`Unable to select new secondary given current healthy Content Nodes=[${healthyNodes}]`)
-
-    const currentHealthySecondary = unhealthyReplicasSet.has(secondary1) ? secondary2 : secondary1
+    let newReplicaNodes = Array.from(newReplicaNodesSet)
+    // We need a minimum of `NUMBER_OF_REPLICA_SET_NODES` to perform the reconfig op. In the case that
+    // the max number replica set attempts were made, push null
+    while (newReplicaNodes.length < NUMBER_OF_REPLICA_SET_NODES) {
+      newReplicaNodes.push(null)
+    }
 
-    // NOTE: For v0 implementation, `newSecondary2` will be the newly selected secondary, whilst the other nodes will be the
-    // original replica set primary and secondary1
-    return {
-      newPrimary: primary,
-      newSecondary1: currentHealthySecondary,
-      newSecondary2: newSecondary
+    // Note: for v0, only issue reconfigs if only 1 secondary is unhealthy. Else, just log out the tentative
+    // new replica set to be assigned.
+
+    if (unhealthyReplicasSet.size === 1) {
+      if (unhealthyReplicasSet.has(primary)) {
+        // If the primary is unhealthy, select the higher clock value of the two secondaries as the new primary,
+        // and then seelct 2 new secondaries
+        const secondariesToClockMap = await this.fetchClockValues(secondary1, secondary2, wallet)
+        newPrimary = secondariesToClockMap[secondary1] >= secondariesToClockMap[secondary2] ? secondary1 : secondary2
+        newSecondary1 = newReplicaNodes[0] // sync
+        newSecondary2 = newReplicaNodes[1] // sync
```

Comment:
this logic seems wrong? the primary would be the secondary with highest clock value, but the `newSecondary1` should just be the other secondary, with the lower clock value, and we only need to elect `newSecondary2` right

Reply:
yeah, this is a good callout. updating